build_projects/shared-build-targets-utils/Utils/DebPackageCreator.cs

```diff
@@ -0,0 +1,197 @@
+            }
+            Directory.CreateDirectory(layoutDirectories.LayoutDirectory);
+
+            Directory.CreateDirectory(layoutDirectories.AbsolutePlacement);
+            Directory.CreateDirectory(layoutDirectories.PackageRoot);
+            Directory.CreateDirectory(layoutDirectories.Samples);
+            Directory.CreateDirectory(layoutDirectories.Docs);
+        }
+
+        private void CopyFilesToDebianLayout(
+            DebianLayoutDirectories layoutDirectories,
+            string debianConfigFile,
+            string inputBinariesDirectory,
+            string manpagesDirectory)
+        {
+            FS.CopyRecursive(inputBinariesDirectory, layoutDirectories.PackageRoot);
+
+            if (manpagesDirectory != null)
+            {
+                FS.CopyRecursive(manpagesDirectory, layoutDirectories.Docs);
+            }
+
+            File.Copy(debianConfigFile,
+                Path.Combine(layoutDirectories.LayoutDirectory, s_debianConfigJsonFileName));
+        }
+
+        private void ReplaceDebianConfigJsonVariables(
+            DebianLayoutDirectories debianLayoutDirectories, 
+            Dictionary<string, string> debianConfigVariables)
+        {
+            var debianConfigFile = Path.Combine(debianLayoutDirectories.LayoutDirectory, s_debianConfigJsonFileName);
+            var debianConfigFileContents = File.ReadAllText(debianConfigFile);
+
+            foreach (var variable in debianConfigVariables)
+            {
+                var variableToken = $"%{variable.Key}%";
+                debianConfigFileContents.Replace(variableToken, variable.Value);
+            }
+
+            File.WriteAllText(debianConfigFile, debianConfigFileContents);
+        }
+
+        private void CreateDebianPackage(
+            DebianLayoutDirectories debianLayoutDirectories,
+            string intermediateDirectory,
+            string outputFile,
+            string packageName,
+            string packageVersion)
+        {
+            var outputDir = Path.Combine(intermediateDirectory, "output");
+            if (Directory.Exists(outputDir))
+            {
+                Directory.Delete(outputDir);
+            }
+            Directory.CreateDirectory(outputDir);
+
+            _dotnet.Exec("deb-tool",
+                "-i", debianLayoutDirectories.LayoutDirectory,
+                "-o", outputDir,
+                "-n", packageName,
+                "-v", packageVersion)
+                .WorkingDirectory(_consumingProjectDirectory)
+                .Execute()
+                .EnsureSuccessful();
+
+            var debianFile = Directory.EnumerateFiles(outputDir, "*.deb").First();
+            File.Copy(debianFile, outputFile, true);
+        }
+
+        private void InitializeDotnetDebTool()
+        {
+            CreateAndRestoreToolConsumingProject();
+        }
+
+        private void CreateAndRestoreToolConsumingProject()
+        {
+            FS.Mkdirp(_consumingProjectDirectory);
```

Comment:
Looks like that code is happening in the 2 places where this is called.  Maybe we can move them both in here.